### PR TITLE
Use Helm chart to spawn worker

### PR DESCRIPTION
**Pull request by marcellp**

### CODE REVIEW

File: src/api/general-services/work-submit/create-worker-k8s.js [hunk 1]
```diff
@@ -1,110 +1,69 @@
...
+      workQueueName,
+      storageSize: '10Gi',
+    };
+  });
+};
+
 const createWorkerResources = async (service) => {
-  const namespaceName = 'worker-refs-heads-master';
+  const { workerHash } = service;
+  const HELM_BINARY = '/usr/local/bin/helm';
```
Comment (marcellp) [inline src/api/general-services/work-submit/create-worker-k8s.js:38]: install helm on the container

File: src/api/general-services/work-submit/index.js [hunk 2]
```diff
@@ -80,13 +80,13 @@ class WorkSubmitService {
...
+   */
   async createWorker() {
-    if (config.clusterEnv === 'development') {
-      logger.log('Not creating a worker because we are running locally...');
-      return;
-    }
+    // if (config.clusterEnv === 'development') {
+    //   logger.log('Not creating a worker because we are running locally...');
+    //   return;
+    // }
```
Comment (marcellp) [inline src/api/general-services/work-submit/index.js]: Uncomment this.

**Review by ivababukova — APPROVED** [review]

File: src/api/general-services/work-submit/create-worker-k8s.js [hunk 3]
```diff
@@ -1,110 +1,69 @@
...
+  // Download value template from Git repository. Fill in needed things.
+  const instanceConfig = await constructChartValues(service);
+  const { name } = tmp.fileSync();
+  await fs.writeFile(name, YAML.stringify(instanceConfig));
 
-  const k8sCoreApi = kc.makeApiClient(k8s.CoreV1Api);
-  const k8sBatchApi = kc.makeApiClient(k8s.BatchV1Api);
+  // Download the chart from the worker repository.
+  const custom = new Downloader({
+    github: { auth: '64ed5ec7a15c641069525e5c43464ead8bbdc20b' },
```
Comment (marcellp) [inline src/api/general-services/work-submit/create-worker-k8s.js]: this needs to be auto generated

Comment (ivababukova) [reply src/api/general-services/work-submit/create-worker-k8s.js]: did you manage to do it automatically or did you end up hardcoding the token?

Comment (marcellp) [reply src/api/general-services/work-submit/create-worker-k8s.js]: i did a mix of things -- i ended up making it an org-wide secret that people have to update manually, but i can put it in the IAC repo or add a script for it later.

for now i will add some docs to iac to remind people to do it.

**Pull request merged**